### PR TITLE
Add PHP parser by ichiriac

### PR DESCRIPTION
**Pull request by pornel**

### CODE REVIEW

File: website/src/parsers/php/php-parser.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import defaultParserInterface from '../utils/defaultParserInterface';
+import pkg from 'php-parser/package.json';
+
+const ID = 'php-parser';
+
+const defaultOptions = {
+  parser: {
+    extractDoc: true,
```
Comment (RReverser) [inline website/src/parsers/php/php-parser.js]: Should this be configurable in the UI?

File: website/src/parsers/php/php-parser.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  loadParser(callback) {
+    require(['php-parser'], callback);
+  },
+
+  parse(Engine, code, options) {
+    const parser = new Engine({...defaultOptions, ...options});
+    return parser.parseCode(code, '');
+  },
+
+  getNodeName(node) {
```
Comment (RReverser) [inline website/src/parsers/php/php-parser.js]: Please implement `nodeToRange` for node highlighting and "jumps" to work.

File: website/src/parsers/php/php-parser.js (hunk 1 above)
Comment (pornel) [inline website/src/parsers/php/php-parser.js]: This adds doc comments as ast nodes, so it doesn't make a big difference.

File: website/src/parsers/php/php-parser.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  homepage: pkg.homepage,
+  locationProps: new Set(['loc']),
+
+  loadParser(callback) {
+    require(['php-parser'], callback);
+  },
+
+  parse(Engine, code, options) {
+    const parser = new Engine({...defaultOptions, ...options});
+    return parser.parseCode(code, '');
```
Comment (RReverser) [inline website/src/parsers/php/php-parser.js]: Does error line highlighting work? If it doesn't currently, please see https://github.com/fkling/astexplorer/pull/247#issuecomment-310726688 for some tips on how to add that.

Comment (pornel) [inline website/src/parsers/php/php-parser.js]: Error highlighting works.

File: website/src/parsers/php/php-parser.js (hunk 2 above)
Comment (pornel) [inline website/src/parsers/php/php-parser.js]: Done

File: website/src/parsers/php/php-parser.js (hunk 3 above)
Comment (RReverser) [inline website/src/parsers/php/php-parser.js]: Great, thanks for checking.

**Comment by fkling** [comment]
Seems fine to me!

**Comment by fkling** [comment]
It throws an error in the production build because uglify is renaming properties. Have to investigate how to fix that.

**Comment by fkling** [comment]
Even if I turn mangling off it throws an error:
screen shot 2017-07-19 at 10 27 47 pm
It seems that `this` is defined as 

screen shot 2017-07-19 at 10 28 34 pm

Not sure what exactly the minifier does that makes the `variable` property not defined.

**Comment by fkling** [comment]
Disabling minification for anything file matching `/php-parser/` works for now...

**Comment by RReverser** [comment]
> because uglify is renaming properties

Wait what? It doesn't do that by default, and I don't remember us configuring it...

**Comment by fkling** [comment]
@RReverser: See the following comments 😉 I assumed it was properties, because that's how it looked like first. But even with mangle turned off completely it's still broken.

**Pull request merged**